snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -333,10 +333,14 @@ def ignore(directory, files):
     def mark_build_done(self):
         build_properties = self.plugin.get_build_properties()
         plugin_manifest = self.plugin.get_manifest()
+        machine_manifest = self._get_machine_manifest()
 
         self.mark_done('build', states.BuildState(
-            build_properties, self._part_properties,
-            self._project_options, plugin_manifest))
+            build_properties, self._part_properties, self._project_options,
+            plugin_manifest, machine_manifest))
+
+    def _get_machine_manifest(self):
+        return {'uname': common.run_output(['uname', '-a'])}
```

Comment:
I'm wondering how flexible this is going to be in the future. Would it be worth going for separate fields like `kernel-name`, `kernel-version` and so on?

Reply:
I asked the security team in the forum if uname -a was appropriate. No reply so far, but pretty easy to change later.

Reply:
Having the hostname in there is somewhat identifying, and seemingly unnecessary to track. I suggest we use e.g. `uname -srvmpio` instead.